orca-web/src/main/groovy/com/netflix/spinnaker/orca/controllers/OperationsController.groovy

```diff
@@ -470,4 +478,58 @@ class OperationsController {
       pipeline.origin = AuthenticatedRequest.spinnakerUserOrigin.orElse('unknown')
     }
   }
+
+  private void addStageAuthorizedRoles(def request, Map pipeline) {
+
+    def jsonObject = jsonSlurper.parseText(request)
+    if (jsonObject.application) {
+      Application application = front50Service.get(jsonObject.application)
+      if (application) {
+        def username = AuthenticatedRequest.getSpinnakerUser().orElse("")
+        def permissions = objectMapper.writeValueAsString(application.getPermission().permissions)
+        def permJsonObject = jsonSlurper.parseText(permissions)
```

Comment:
More serde hoops that I don't think are necessary. If you want `permissions` as a simple map, you could do `objectMapper.convertValue(permissions, new TypeReference<Map<String, Object>>() {})` or similar and skip the serde hoops entirely.

Reply:
Incorporated the change.